imap_processing/spice/geometry.py

```diff
@@ -1 +1,128 @@
-"""Functions for computing geometry using SPICE."""
+"""Functions for computing geometry, many of which use SPICE."""
+
+from collections.abc import Iterable
+from enum import Enum
+from typing import NamedTuple, Optional, Union
+
+import numpy as np
+import spiceypy as spice
+
+from imap_processing.spice.kernels import ensure_spice
+
+
+class SpiceId(NamedTuple):
+    """Class that represents a unique identifier in the NAIF SPICE library."""
+
+    strid: str
+    numid: int
+
+
+class SpiceBody(Enum):
+    """Enum containing SPICE IDs for bodies that we use."""
+
+    # A subset of IMAP Specific bodies as defined in imap_wkcp.tf
+    IMAP = SpiceId("IMAP", -43)
+    IMAP_SPACECRAFT = SpiceId("IMAP_SPACECRAFT", -43000)
+    # IMAP Pointing Frame (Despun) as defined in iamp_science_0001.tf
+    IMAP_DPS = SpiceId("IMAP_DPS", -43901)
+    # Standard NAIF bodies
+    SSB = SpiceId("SOLAR_SYSTEM_BARYCENTER", 0)
+    SUN = SpiceId("SUN", 10)
+    EARTH = SpiceId("EARTH", 399)
+
+
+class SpiceFrame(Enum):
+    """Enum containing SPICE IDs for reference frames, defined in imap_wkcp.tf."""
+
+    # Standard SPICE Frames
+    J2000 = SpiceId("J2000", 1)
+    ECLIPJ2000 = SpiceId("ECLIPJ2000", 17)
+    # IMAP specific
+    IMAP_SPACECRAFT = SpiceId("IMAP_SPACECRAFT", -43000)
+    IMAP_LO_BASE = SpiceId("IMAP_LO_BASE", -43100)
+    IMAP_LO_STAR_SENSOR = SpiceId("IMAP_LO_STAR_SENSOR", -43103)
+    IMAP_LO = SpiceId("IMAP_LO", -43105)
+    IMAP_HI_45 = SpiceId("IMAP_HI_45", -43150)
+    IMAP_HI_90 = SpiceId("IMAP_HI_90", -43160)
+    IMAP_ULTRA_45 = SpiceId("IMAP_ULTRA_45", -43200)
+    IMAP_ULTRA_90 = SpiceId("IMAP_ULTRA_90", -43210)
+    IMAP_MAG = SpiceId("IMAP_MAG", -43250)
+    IMAP_SWE = SpiceId("IMAP_SWE", -43300)
+    IMAP_SWAPI = SpiceId("IMAP_SWAPI", -43350)
+    IMAP_CODICE = SpiceId("IMAP_CODICE", -43400)
+    IMAP_HIT = SpiceId("IMAP_HIT", -43500)
+    IMAP_IDEX = SpiceId("IMAP_IDEX", -43700)
+    IMAP_GLOWS = SpiceId("IMAP_GLOWS", -43750)
+
+
+def imap_state(
+    et: Union[np.ndarray, float],
+    ref_frame: Optional[SpiceFrame] = None,
+    observer: Optional[SpiceBody] = None,
```

Comment:
You can put the defaults directly in the call.
```suggestion
    ref_frame: SpiceFrame = SpiceFrame.ECLIPJ2000,
    observer: SpiceBody = SpiceBody.SUN,
```

Reply:
Hahaha... good point